blog/content/post/2021-02-23-OSF_vs_OCP_OSF.md

```diff
@@ -0,0 +1,119 @@
+
+*Source: IDC Worldwide Open Compute Project Compute and Storage Infrastructure Market Forecast, May 2020.*
+
+The OCP Open System Firmware (OSF) project is aiming to allow system firmware
+like [BIOS][3] and [BMC][4] to be modified and their source code shared with
+the owners of OCP-certified hardware. Since March 2021, supporting OSF is
+mandatory for servers with OCP badging. The ecosystem partners have decided to
+implement such open-source firmware as coreboot for their OSF architecture,
+since having the source helps to adapt the firmware to the needs of even the
+most demanding clients by fixing the firmware bugs and expanding the device's
+feature set - at a reduced cost - which allows making the most out of both
+your existing and future products by extending their lifecycle and bringing them
+to the new previously unthought-of markets.
+
+![Embedded software market](/img/Embedded_Software_Market.png)
+
+Just being open-source - by itself is not enough for the firmware to meet the
+OCP certification: the extra requirements are mentioned in a [OSF Checklist][5].
+The source code should be easily buildable by a native (ideally the open-source)
+toolchain and the flashing utility should be either open-source like [flashrom][6]
+or at least documented well enough for an open-source implementation to be
+written. The key OSF components have to be open-source, and although it is
+allowed to include some closed-source ones *(aka binary blobs)* to OSF - there
+should be a valid justification for them to be approved, such as "containing
+silicon IP": i.e. a blob has been provided by your SV (Silicon Vendor) like
+Intel to set up their chip during boot, which is needed for your device to run.
+And all the documentation should be good enough and distributable without NDAs.
+
+![OSF features](/img/OSF_features.png)
+
+The diagram above shows 6 types of OSF features: the fundamental features to the
+more advanced ones are depicted from left to right and 1st to 3rd are essential.
+
+1. Chip Initialization and Integration: the firmware components for the initial
+setup of the motherboard's key chips - like the chips themselves - are usually
+provided by the Silicon Vendors, sometimes in a binary-only form *(without the*
+*source code)* like Intel FSP and later AMD AGESA. Some binary-only components,
+depending on their complexity, could be reverse-engineered by us at [3mdeb][7] to
+create a higher quality opensource replacement with less bugs and more features.
+
+2. Board Customization: in addition to the chip initialization, there should
+be a board-specific setup for the things like GPIO, PCIe bifurcation and BMC
+interface. In a coreboot, these could be configured at the mainboard files.
+
+3. System Table Generation: system tables like SMBIOS and ACPI tables have to
+be generated by OSF to provide access to various management features for an OS.
```

Comment:
@mikebdp2 what is the meaning of OSF here?

Reply:
@pietrushnic While writing I meant the open-source but any firmware has to do this, - and anyways the discrepancy between the Open-Source Firmware and Open System Firmware terms should go away after March 2021 _(although not instantly, since there still would be the older OCP-certified products the firmware of which wasn't open-source)_. So I don't think I should expand the OSF term at every mention - otherwise this blog post could become too wordy.